src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1705,6 +1710,28 @@ public function isXmlHttpRequest()
         return 'XMLHttpRequest' == $this->headers->get('X-Requested-With');
     }
 
+    /**
+     * Checks whether the client browser prefers safe content or not according to RFC8674.
+     *
+     * @see https://tools.ietf.org/html/rfc8674
+     */
+    public function preferSafeContent(): bool
+    {
+        if (null !== $this->isSafeContentPreferred) {
+            return $this->isSafeContentPreferred;
+        }
+
+        if (!$this->isSecure()) {
+            $this->isSafeContentPreferred = false;
+
+            return $this->isSafeContentPreferred;
+        }
+
+        $this->isSafeContentPreferred = \in_array('safe', (AcceptHeader::fromString($this->headers->get('Prefer'))->all()), false);
```

Comment:
extra brackets should be removed

Reply:
why "false" btw? true would be more correct, isn't it

Reply:
I was not sure whether the spec defines case sensitive values or not